src/v2/kuadrant/pipeline/tasks/mod.rs

```diff
@@ -7,20 +7,21 @@ use crate::v2::services::Service;
 use std::rc::Rc;
 
 #[allow(dead_code)]
-trait Task {
+pub trait Task {
     fn apply(self: Box<Self>, ctx: &mut ReqRespCtx) -> TaskOutcome;
 }
 
 #[allow(dead_code)]
-struct PendingTask {
+pub struct PendingTask {
     is_blocking: bool,
     allow_task: Option<Box<dyn Task>>,
```

Comment:
This will more than likely become a vec of `allow_tasks` to perform, `deny_task` _might_ also but I haven't thought through all the scenarios yet